harness/determined/cli/proxy.py

```diff
@@ -221,7 +221,10 @@ def handle(self) -> None:
             c1.join()
             c2.join()
 
-    return socketserver.ThreadingTCPServer((tunnel.local_addr, tunnel.local_port), TunnelHandler)
+    class ReuseAddrServer(socketserver.ThreadingTCPServer):
+        allow_reuse_address = True
```

Comment:
This needs to be limited to non-windows, since windows' implementation of SO_REUSEADDR is basically an "open a security hole" switch.

From the (overly detailed) [windows docs](https://learn.microsoft.com/en-us/windows/win32/winsock/using-so-reuseaddr-and-so-exclusiveaddruse#application-strategies):

> The SO_REUSEADDR option has very few uses in normal applications aside from multicast sockets where data is delivered to all of the sockets bound on the same port. Otherwise, any application that sets this socket option should be redesigned to remove the dependency since it is eminently vulnerable to "socket hijacking".

(To Windows' credit, their non-SO_REUSEADDR behavior is a better default than the linux non-SO_REUSEADDR behavior.)

Example script for playing around with this on different OSs:

```python
import socket
import sys
s = socket.socket()
if "reuse" in sys.argv:
    print("SO_REUSEADDR")
    s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
if "exclusive" in sys.argv:
    print("SO_EXCLUSIVEADDRUSE")
    s.setsockopt(socket.SOL_SOCKET, socket.SO_EXCLUSIVEADDRUSE, 1)
s.bind(("127.0.0.1", 1991))
s.listen()
print("success")
input()
```

Reply:
disabled on windows.